pkg/nodes/enricher/enricher_impl.go

```diff
@@ -101,6 +101,11 @@ func (e *enricherImpl) enrichWithScan(node *storage.Node, nodeInventory *storage
 	}
 
 	for _, scanner := range scanners {
+		// Prevent unnecessary scanner calls - v4 only works on indexReports, v2/Clairify only on nodeInventory.
+		if (scanner.GetNodeScanner().Type() == types.ScannerV4 && nodeInventory != nil) ||
+			(scanner.GetNodeScanner().Type() == types.Clairify && indexReport != nil) {
+			continue
+		}
```

Comment:
Fully optional, but it makes it easier to read ;)
```suggestion
		if (scanner.GetNodeScanner().Type() == types.ScannerV4 && indexReport == nil) ||
			(scanner.GetNodeScanner().Type() == types.Clairify && nodeInventory == nil) {
			continue
		}
```